CMakeLists.txt

```diff
@@ -30,9 +31,42 @@ endif(NOT DEFINED WANT_EIO)
 aux_source_directory("${CMAKE_CURRENT_SOURCE_DIR}/src" EVFIBERS_SOURCES)
 aux_source_directory("${CMAKE_CURRENT_SOURCE_DIR}/coro" CORO_SOURCES)
 
+# OSX-related checks - linking, using ucontext, etc. is slightly different than
+# on linux
+if(APPLE)
+	# def needed to get ucontext support
+	set(CMAKE_REQUIRED_DEFINITIONS "${CMAKE_REQUIRED_DEFINITIONS} -D_XOPEN_SOURCE")
+	add_definitions(-D_XOPEN_SOURCE)
+	# ucontext is labeled deprecated, so make sure -Werror doesn't kill the
+	# compile
+	set(CMAKE_C_FLAGS "-Wno-error=deprecated-declarations ${CMAKE_C_FLAGS}")
+	# OSX uses MAP_ANON for mmap while linux uses MAP_ANONYMOUS
+	set(FBR_MAP_ANON_FLAG MAP_ANON)
+	set(EIO_LD_WHOLE_ARCHIVE -Wl,-all_load)
+	set(EIO_LD_NOWHOLE_ARCHIVE )
+else(APPLE)
+	set(FBR_MAP_ANON_FLAG MAP_ANONYMOUS)
+	set(EIO_LD_WHOLE_ARCHIVE -Wl,-whole-archive)
+	set(EIO_LD_NOWHOLE_ARCHIVE -Wl,-no-whole-archive)
+endif(APPLE)
+
+# use of swapcontext hits a false positive stack overread. Recent versions of
+# clang have stack-protector on by default which triggers it, so disable
+CHECK_C_COMPILER_FLAG("-fno-stack-protector" FBR_HAS_NO_STACK_PROTECT)
+if(FBR_HAS_NO_STACK_PROTECT)
+    set(CMAKE_C_FLAGS "-fno-stack-protector ${CMAKE_C_FLAGS}")
```

Comment:
I've forgotten exactly when the errors occur over the last week :), so I'll rerun with various permutations on mac and linux boxes and see what happens.